src/engine/renderer/tr_bsp.cpp

```diff
@@ -447,6 +447,11 @@ R_LoadLightmaps
 static const int LIGHTMAP_SIZE = 128;
 static void R_LoadLightmaps( lump_t *l, const char *bspName )
 {
+	if ( r_vertexLighting->integer )
```

Comment:
does vertex lighting really mean lightmapping is disabled?

Reply:
Yes. This cvar is really misnamed and the way it is used is really confusing:

- everywhere in engine (and game UI) the purpose of this cvar is to know if light mapping is enabled or not;
- engine never does vertex lighting but grid lighting instead.

I plan to make a PR to fix this later (later to not break any possible work-in-progress branches).

In the future I want to have this:

- `r_lightMapping 0`: light mapping disabled, use grid lighting for everything (map and models)
- `r_lightMapping 1`: light mapping enabled, use light map anytime a surface has one (map), use grid light for other surfaces (models)

This naming and value scheme would not only make code easier to read, but it would also enable alternative light scenario like:

- `r_lightMapping 2`: raytrace the light map at map load.
- `r_lightMapping 3`: raytrace the light in real time.

Or anything else we can imagine or cannot imagine yet.